### PR TITLE
Update CmdStan & CmdStanPy groups

### PR DESCRIPTION
**Pull request by ahartikainen**
## Description

New groups for CmdStan and CmdStanPy:

### CmdStan
- [x] Predictions
- [x] Constant data
- [x] Predictions constant data
- [x] Log Likelihood

### CmdStanPy
- [x] Predictions
- [x] Constant data
- [x] Predictions constant data
- [x] Log Likelihood

## Checklist

- [x] Does the PR follow [official](https://github.com/arviz-devs/arviz/blob/master/CONTRIBUTING.md#pull-request-checklist) 
      PR format?
- [ ] Is the new feature properly documented with an example?
- [x] Does the PR include new or updated tests to cover the new feature (using [pytest fixture pattern](
      https://docs.pytest.org/en/latest/fixture.html#fixture))?
- [x] Is the code style correct (follows pylint and black guidelines)?
- [ ] Is the change listed in [changelog](https://github.com/arviz-devs/arviz/blob/master/CHANGELOG.md#v0xx-unreleased )?

### CODE REVIEW

**Comment by ahartikainen** [comment]
@OriolAbril I can update the examples for CmdStan and CmdStanPy after this PR.

File: arviz/data/io_cmdstan.py [hunk 1]
```diff
@@ -690,21 +794,34 @@ def from_cmdstan(
...
+    constant_data_var : str, List[str]
+        Variable(s) used for slicing constant_data. If not defined, all
+        data variables are imported.
+    predictions_constant_data : str
+        Constant data for predictions used in the sampling.
+        Path to data file in Rdump format.
+    predictions_constant_data_var : str, List[str]
+        Variable(s) used for slicing predictions_constant_data.
+        If not defined, all data variables are imported.
     log_likelihood : str
```
Comment (OriolAbril) [inline arviz/data/io_cmdstan.py]: I think this should now be str or list of str

File: arviz/data/io_cmdstan.py [hunk 2]
```diff
@@ -352,6 +387,70 @@ def observed_data_to_xarray(self):
...
+        else:
+            if isinstance(log_likelihood, str):
+                log_likelihood = [log_likelihood]
+            log_likelihood_cols = [
+                col for col in columns if any(item == col.split(".")[0] for item in log_likelihood)
+            ]
+            data = _unpack_dataframes([item[log_likelihood_cols] for item in self.posterior])
+        return dict_to_dataset(data, coords=self.coords, dims=self.dims)
+
     def to_inference_data(self):
```
Comment (OriolAbril) [inline arviz/data/io_cmdstan.py]: still to be updated

File: arviz/tests/test_data_cmdstan.py [hunk 3]
```diff
@@ -173,11 +187,15 @@ def test_inference_data_input_types2(self, paths, observed_data_paths):
             )
             test_dict = {
                 "posterior": ["mu", "tau", "theta_tilde", "theta"],
+                "posterior_predictive": ["y_hat"],
+                "predictions": ["y_hat"],
                 "prior": ["mu", "tau", "theta_tilde", "theta"],
                 "prior_predictive": ["y_hat"],
-                "sample_stats": ["log_likelihood"],
+                "sample_stats": ["divergent"],
```
Comment (OriolAbril) [inline arviz/tests/test_data_cmdstan.py]: "diverging" after renaming

File: arviz/tests/test_data_cmdstan.py [hunk 4]
```diff
@@ -200,17 +218,22 @@ def test_inference_data_input_types3(self, paths, observed_data_paths):
...
+                coords={
+                    "school": np.arange(8),
+                    "log_lik_dim_0": np.arange(8),
+                    "y_hat": np.arange(8),
+                },
                 dims={"theta": ["school"], "y": ["school"], "y_hat": ["school"], "eta": ["school"]},
             )
             test_dict = {
                 "posterior": ["mu", "tau", "theta_tilde", "theta"],
+                "sample_stats": ["divergent"],
```
Comment (OriolAbril) [inline arviz/tests/test_data_cmdstan.py]: diverging

**Comment by ahartikainen** [comment]
Great, I will fix these tomorrow (or anyone from dev team is free to fix this too)

**Comment by OriolAbril** [comment]
I was also thinking about the different names for matching variables, and I wanted to ask, would it be feasible to export the dict input in from_pystan's log_likelihood to other args like posterior_predictive and maybe other converters like from_cmdstanpy? 

Always allowing the list option plus `data_pairs` arguments, and extending to allowing same variable names, it would make the import more lengthy but afterwards it would simplify plotting. We could create an issue to discuss the details.

**Comment by ahartikainen** [comment]
I think we need to think how to use names and which names to use.

For example we could have specific default names for InferenceData and add the transformation to attributes or do the change in functions etc. Let's create another issue for this.

I will change the changelog and merge after.

**Pull request merged**